tools/bcr_validation.py

```diff
@@ -164,18 +170,78 @@ def verify_presubmit_yml_change(registry, module_name, version):
       validation_results.append((BcrValidationResult.GOOD, "The presubmit.yml file matches the previous version."))
   return validation_results
 
-def validate_module(registry, module_name, version):
-  print_expanded_group(f"Validating {module_name}@{version}")
+def apply_patch(work_dir, patch_strip, patch_file):
+  # Requries patch to be installed
+  subprocess.run(
+      ["patch", "-p%d" % patch_strip, "-l", "-i", patch_file], shell=False, check=True, env=os.environ, cwd=work_dir
+  )
 
+def remove_trailing_empty_lines(lines):
```

Comment:
just curious -- why is this necessary?

Reply:
That part is not very important, but fixing the line endings is important, otherwise there will be malformed diff result. Done.